request/request.go

```diff
@@ -1,65 +1,148 @@
+//
+//   - trusted_ips
+//   - real_ip_header
+//   - real_ip_recursive
+//
+// Note: we do not currently offer support for Forwarded HTTP Extension (RFC 7239)
+// since it is not supported by ngx_http_realip_module.
 func (r Request) GetForwardedPort() (port string, err error) {
 	return r.AskString(`kong.request.get_forwarded_port`)
 }
 
+// kong.Request.GetHttpVersion() returns the HTTP version
+// used by the client in the request, returning values
+// such as "1"", "1.1", "2.0", or nil for unrecognized values.
 func (r Request) GetHttpVersion() (version string, err error) {
 	return r.AskString(`kong.request.get_http_version`)
 }
 
+// kong.Request.GetMethod() returns the HTTP method of the request.
+// The value is normalized to upper-case.
 func (r Request) GetMethod() (m string, err error) {
 	return r.AskString(`kong.request.get_method`)
 }
 
+// kong.Request.GetPath() returns the path component of the request’s URL.
+// It is not normalized in any way and does not include the querystring.
 func (r Request) GetPath() (string, error) {
 	return r.AskString(`kong.request.get_path`)
 }
 
+// kong.Request.GetPathWithQuery() returns the path, including
+// the querystring if any. No transformations/normalizations are done.
 func (r Request) GetPathWithQuery() (string, error) {
 	return r.AskString(`kong.request.get_path_with_query`)
 }
 
+// kong.Request.GetRawQuery() returns the query component of the request’s URL.
+// It is not normalized in any way (not even URL-decoding of special characters)
+// and does not include the leading ? character.
 func (r Request) GetRawQuery() (string, error) {
 	return r.AskString(`kong.request.get_raw_query`)
 }
 
+// kong.Request.GetQueryArg() returns the value of the specified argument,
+// obtained from the query arguments of the current request.
+//
+// The returned value is either a string, a boolean true if
+// an argument was not given a value, or nil if no argument with name was found.
+//
+// If an argument with the same name is present multiple times in the querystring,
+// this function will return the value of the first occurrence.
 func (r Request) GetQueryArg() (string, error) {
 	return r.AskString(`kong.request.get_query_arg`)
 }
 
+// kong.Request.GetQuery() returns a map of query arguments
+// obtained from the querystring. Keys are query argument names.
+// Values are either a string with the argument value, a boolean true
+// if an argument was not given a value, or an array if an argument
+// was given in the query string multiple times. Keys and values are
+// unescaped according to URL-encoded escaping rules.
+//
+// Note that a query string `?foo&bar` translates to two boolean true arguments,
+// and ?foo=&bar= translates to two string arguments containing empty strings.
+//
+// The max_args argument specifies the maximum number of returned arguments.
+// Must be greater than 1 and not greater than 1000, or -1 to specify the
+// default limit of 100 arguments.
```

Comment:
this return type is really uncomfortable to use on Go.  I think values should always be an array of strings:

	?foo&bar =>  {"foo": [], "bar": []}
	?foo=&bar= => {"foo": [""], "bar": [""]}